### DISCUSSION TITLE
dde 在 NixOS 中移植的问题及追踪

### DISCUSSION THREAD

**Discussion by wineee** [martyr-deepin/dde-nixos#1] [Show and tell]
## 可能相关的 issues
- [x] https://github.com/linuxdeepin/developer-center/issues/3167
- [x] https://github.com/linuxdeepin/developer-center/issues/3155
- [x] https://github.com/linuxdeepin/release (待完善)

## pr 跟踪

移到 https://github.com/linuxdeepin/dde-nixos/discussions/3

### 1. 安装路径

见 issues 1

### 2. 代码路径

1. 代码中 /usr/bin/xxx , /usr/sbin/xxx -> xxx，是否存在副作用？
2. 代码中 /usr/lib/deepin-daemon/dde-warning-dialog, lib 目录不再 PATH 中，不能直接换成 dde-warning-dialog，如果换成 dde-daemon 的路径 ${dde-daemon}/lib/deepin-daemon/dde-warning-dialog，会引入dde-daemon作为构建依赖，即使编译不需要它。
4. Shebangs: `#!/usr/bin/python` -> `#!/usr/bin/env python` ,  `#!/bin/bash` -> `#!/usr/bin/env bash`
5. 代码中 /bin/sh  变成 ${runtimeShell}

- 代码中 /etc 路径，一部分在 $out/etc（本项目文件）， 一部分在 /etc， 一部分在 /run/current-system/sw/etc，一部分不知道在哪
- /usr/share，一部分在 $out/share， 一部分在 /etc， 一部分在 /run/current-system/sw/share

找不着的文件
- [x] /usr/share/X11/xkb/rules/base.xml
- [ ] 待更新

### 3. 杂项文件中的路径  
-  .desktop 文件，dbus / systemd services 文件中， /usr/bin/xxx , /usr/sbin/xxx -> xxx

**Comment by wineee** [comment]
### TODO
- [x] dconfig  挂载 /usr/dsg 目录，修改读取路径？，写 /var 权限？
- [x] gsettings-schemas...
- [x] polkit-1/actions/com.***.policy
- [x] 确保 lib/pkgconfig， lib/cmake 文件内容正确
 
- [x] deepin-voice-note 无法录音
- [x] deepin-picker  无法跟随鼠标更新
- [x] dde-control-center 
- [x] deepin-screen-recorder 无法录屏
- [x] dde-launcher-wapper: com.deepin.SessionManager 找不到
- [x] dde-dock 找不到 org.deepin.dde.dock（不是在 deepin-desktop-schemas ？）
...

- [x] 修理插件打包
- [x] nixosModules 功能拆分

**Comment by gnuhub** [comment]
@wineee 期待在nixos使用dde

**Reply by wineee** [reply]
@gnuhub 现在可以用了，有兴趣试试吗
